studentSubmissions/script.js

```diff
@@ -0,0 +1,162 @@
+// console.log("Find Submission by Name", submissions);  
+// Had troubles logging this one...
+
+// specSeven
+// Declare a function named findLowestScore
+// Parameter(s): array
+// Functionality: return the object in the `array` that has the lowest score.
+
+const findLowestScore = (array) => {
+    let lowestScore = 100;
+    array.forEach((sub) => {
+        const lowerScore = sub.score <= lowestScore ? true : false;
+        // console.log(lowerScore) testing if the forEach iterator works
+        if (lowerScore === true) {
+            lowestScore = sub.score
+        }
+    });
+    return lowestScore;
+}
+findLowestScore(submissions)
+console.log("Lowest Score:", findLowestScore(submissions));
+// halfway done, gotta return something look up let if ex. on w3s this comment was on line 77
+
+// specEight
+// Declare a function named findAverageScore
+// Parameter(s): array
+// Functionality: return the average quiz score.
+
+const findAverageScore = (array) => {
+    let totalScore = 0;
+    array.forEach((sub)=> {
+        totalScore += sub.score;
+    }); 
+    const averageScore = totalScore / array.length;
+    return averageScore
+};
+findAverageScore(submissions)
+console.log("Average Score:", findAverageScore(submissions));
+
+// specNine
+// Declare a function named filterPassing
+// Parameter(s): array
+// Functionality: return a new array using the filter method. The filter method should find objects in the `array` that have passing scores.
+
+const filterPassing = (array) => array.filter((sub) => sub.passed === true);
+filterPassing(submissions);
+console.log("Passing Submissions:", filterPassing(submissions));
+
+// specTen
+// Declare a function named filter90AndAbove
+// Parameter(s): array
+// Functionality: return a new array using the filter method. The filter method should find objects in the `array` that have scores greater than or equal to 90.
+
+const filter90AndAbove = (array) => array.filter((sub) => sub.score >= 90);
+filter90AndAbove(submissions)
+console.log("Submissions 90 and Above:", filter90AndAbove(submissions));
+
+// Extended Challenges Specs
+// extendedSpecOne
+// Create a function named createRange
+// Parameter(s): start, end
+// Functionality: construct and return an array of integers starting with the start parameter and ending at the end parameter. e.g `createRange(2, 5)` returns `[2, 3, 4, 5]`.
+// const nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+const createRange = (start, end) => {
+    const range = [];
+    for (let i = start; i <= end; i++) {
+        range.push(i);
+    }
+    return range;
+};
+const result = createRange(2,5);
+console.log(result);
+
+// extendedSpecTwo
+// Create a function named countElements
+// Parameter(s): array (an array of strings)
+// Functionality: construct and return an object with the array values as keys and the number of times that key appears in the array as values. e.g. `countElements(['a', 'b', 'a', 'c', 'a', 'b'])` returns `{ a: 3, b: 2, c: 1 }`.
```

Comment:
Not sure if you tried this extended challenge. This is a tough one. One possible answer: 

```
const countElements = (array) => {
  const counter = {}; // create an empty object

  for (let elem of array) { // loop over each element of the array
    if (!counter[elem]) {  // if the array element IS NOT a property in our counter object
      counter[elem] = 1;  // create a property named after the array element, and assign it a value of 1
    } else { // if the array element is already a property of the object
      counter[elem]++; // increase that property value by 1
    }
  }
  return counter;
};
```